rir/src/interpreter/interp.cpp

```diff
@@ -1326,7 +1382,8 @@ SEXP evalRirCode(Code* c, Context* ctx, SEXP* env, const CallContext* callCtxt,
             // environment
             memset(&bindingCache, 0, sizeof(bindingCache));
             SEXP e = ostack_pop(ctx);
-            assert(TYPEOF(e) == ENVSXP && "Expected an environment on TOS.");
+            assert((TYPEOF(e) == ENVSXP || LazyEnvironment::check(e)) &&
+                   "Expected an environment on TOS.");
```

Comment:
materialize if lazy env?

Reply:
No, that would materialize lot of environments that are actually not used.